src/context/ModalContext.ts

```diff
@@ -0,0 +1,38 @@
+/// Modal context sets and removes the currently display modal on the page
+// eslint-disable-next-line prettier/prettier
+import React, { createContext, useState } from "react"
+
+type ModalComponentType = React.FC | React.ReactNode | undefined
+type ModalState = Partial<Record<string, ModalComponentType>>
+type OverlayParams = {
+	key: string
+	component: ModalComponentType
+}
+type ModalContext = {
+	setOverlay: (modal?: OverlayParams) => void
+	overlay?: ModalState
+}
+
+export const useModal = (): ModalContext => {
+	const [overlay, setLay] = useState<ModalState | undefined>()
+
+	// setOverlay handles removing and setting the current modal
+	const setOverlay = (data?: OverlayParams) => {
+		if (data) {
+			// eslint-disable-next-line prettier/prettier
+			const { key, component } = data
+			setLay(() => ({
+				[key as string]: component
+			}))
+		} else {
+			setLay(undefined)
+		}
+	}
+
+	return {
+		setOverlay,
+		overlay
+	}
+}
+
+export const ModalContext = createContext<ModalContext>({} as unknown as ModalContext)
```

Comment:
Honestly - I don't like the idea of migrating from regular modal state management via `useState` to setting modals as overlays. This gives us some flexibility in making anything being modal, but on the other hand - brings unnecessary complexity. This is common practice to handle modals not through a global context, but through local state management, and I would like to keep it that way. Most-known UI libraries(Material UI, Reactstrap, Semantic UI, etc.) are handling it exactly the way we have it currently.

@Koochr WDYT?

Reply:
Let's keep it as is. We don't need global Modal context since we don't have "global" modals, every modal is used only in 1 component